autonbox/clean_augment.py

```diff
@@ -0,0 +1,73 @@
+import os
+import pandas as pd
+
+from d3m import container
+from d3m import utils as d3m_utils
+from d3m.metadata import base as metadata_base, hyperparams
+from d3m.primitive_interfaces.base import CallResult
+from d3m.primitive_interfaces.transformer import TransformerPrimitiveBase
+
+import autonbox
+
+Inputs = container.Dataset
+Outputs = container.Dataset
+
+class Hyperparams(hyperparams.Hyperparams):
+    original_columns = hyperparams.Set(
+        elements=hyperparams.Hyperparameter[str](''),
+        default=(),
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/ControlParameter'],
+        description="List of the original columns"
+    )
+    percentage_missing = hyperparams.Uniform(
+        lower=0.0, 
+        upper=1.0, 
+        default=0.3,
+        semantic_types=['https://metadata.datadrivendiscovery.org/types/TuningParameter'],
+        description='Percentage of missing data allowed (above this percentage and the line is deleted)'
+    )
+
+
+class CleanAugmentationPrimitive(TransformerPrimitiveBase[Inputs, Outputs, Hyperparams]):
+    """
+        Remove rows which haven't been augmented following the datamart augmentation 
+        (any rows with more than percentage_missing columns which are resulting from augmentation)
+        
+        NB: This primitive results might reduce the number of row
+    """
+
+    metadata = metadata_base.PrimitiveMetadata({
+        'id': 'fe0f1ac8-1d39-463a-b344-7bd498a31b92',
+        'version': '0.1',
```

Comment:
Use three part naming conversion for version: i.e. [Major].[Minor].[Patch]
`'version':'0.1.0'`